### PR TITLE
Visualisise Betweeness

### PR DESCRIPTION
**Pull request by Gomango999**
This PR implements a tutorial for visualising both vertex and edge betweenness centrality measures in a graph, as  suggested [here](https://github.com/igraph/python-igraph/pull/460#issuecomment-981126320).

betweenness

### CODE REVIEW

**Comment by ntamas** [comment]
We have a handy function named `rescale` in `igraph.utils` -- you can give it a sequence of input values and a desired output range, and it will rescale everything in a linear manner such that the minimum and the maximum lines up with the range that you have given to it. This can be used for determining the widths and sizes.

Also, I was wondering whether we should use a more straightforward graph. [Krackhardt's kite graph](https://www.wikiwand.com/en/Krackhardt_kite_graph) is typically used in textbooks to illustrate differences between various centrality measures, so maybe we could use that here as well?

**Comment by iosonofabio** [comment]
I'd like to show both: a textbook example and a larger, messy graph (for data science folks)

**Comment by vtraag** [comment]
Personally I also like the more messy results.

**Comment by Gomango999** [comment]
Thanks as always for the feedback and tips and tricks :) I've switched to `ig.rescale`, and also included an example for the Krackhardt graph. The results are below:

betweenness

**Comment by iosonofabio** [comment]
I've improved the matplotlib with panel labels and a fixed figure size, apart from that it looks good, merging.

File: doc/source/tutorials/betweenness/betweenness.rst [hunk 1]
```diff
@@ -11,7 +11,7 @@ Visualizing Betweenness
 .. _edge_betweenness: https://igraph.org/python/doc/api/igraph._igraph.GraphBase.html#edge_betweenness
 .. |edge_betweenness| replace:: :meth:`edge_betweenness`
 
-This example will demonstrate how to visualize both vertex and edge betweenness with a custom defined color palette. We use the methods |betweenness|_ and |edge_betweenness|_ respectively.
+This example will demonstrate how to visualize both vertex and edge betweenness with a custom defined color palette. We use the methods |betweenness|_ and |edge_betweenness|_ respectively, and demonstrate the effects on a standard `Krackhardt Kite <https://www.wikiwand.com/en/Krackhardt_kite_graph>`_ graph, as well as a `Barabási-Albert <https://en.wikipedia.org/wiki/Barab%C3%A1si%E2%80%93Albert_model>`_ random graph.
```
Comment (szhorvat) [inline doc/source/tutorials/betweenness/betweenness.rst:14]: will demonstrate -> demonstrates

**Comment by szhorvat** [comment]
I don't know how I feel about the arbitrary non-linear rescaling of betweenness just to "make it look good" ... Whether we want it or not, people do learn network science by using igraph, and I fear that this is teaching a bad practice. What do you think @vtraag ? Maybe do the rescaling for the sizes (which is common: we can scale either the vertex radius or vertex area with some cantrality), but not the colour?

@Gomango999 Can we have a colour bar added to this plot? One of the major benefits that matplotlib brings is easy colour bars. Whatever you do with the non-linear rescaling, the colour bar's labels should show true betweenness values, not rescaled ones.

**Comment by szhorvat** [comment]
You did the rescaling because preferential attachment networks have a very very skewed betweenness distribution. I would recommend a Watts-Strogatz network in 2D, e.g. a 10 by 10 grid, with rewiring probability between 0.05-0.1. This also gives a nice looking network plot, but with a more "normal" betweenness distribution.

image

**Comment by szhorvat** [comment]
Also, it bothers me that there is a variable called `edge_betweenness` that does not actually store edge betweenness, but a non-linearly rescaled quantity.

**Comment by Gomango999** [comment]
Cheers :) You were right that using Barabási-Albert was giving me skewed results, which encouraged me to scale the colours in a weird way so the graph didn't end up looking like just two colours. All these suggestions are completely fair, so I'll implement them when I get the time.

Happy holidays!

**Pull request merged**